rpc/websocket.go

```diff
@@ -282,8 +283,12 @@ type websocketCodec struct {
 	pingReset chan struct{}
 }
 
-func newWebsocketCodec(conn *websocket.Conn, host string, req http.Header) ServerCodec {
-	conn.SetReadLimit(wsMessageSizeLimit)
+func newWebsocketCodec(conn *websocket.Conn, cfg *clientConfig, host string, req http.Header) ServerCodec {
```

Comment:
AFAIK, this function is both invoked by the client and server sides. So the arg `clientConfig` maybe incorrect. I'd advice passing `sizeLimit` instead.

And in the server's side, we use the const `wsMessageSizeLimit` seems is enough, in case of DDoS. If you really need to configure a large number, then it is recommended to pass the value over the command line argument.(but I think 32MB is totally enough)

Reply:
Michal Horvat

Reply:
Michal Horvat


https://github.com/MAJKYTHEBOSS/go-ethereum

Reply:
Michal King Horvat

Reply:
(didn't know spam on github pull requests was a thing... kind of weird....)

Anyway:
I pushed a  change so the message size limit is an argument to `newWebSocketCodec`. I agree that it seems strange to pass a `clientConfig` around inside the server code.

On the other hand,  `clientConfig` is used in some server code like https://github.com/tylerni7/go-ethereum/blob/wsMessageSizeLimit/rpc/server.go#L102 but I suppose that is more to construct a client for the server to use. It also seems nice to have the options struct passed around deeper in case other settings arise which would be nice to configure.